src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt

```diff
@@ -0,0 +1,44 @@
+package com.hibob.academy.filters
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jws
+import io.jsonwebtoken.Jwts
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.container.ContainerRequestFilter
+import org.springframework.stereotype.Component
+import com.hibob.academy.service.SessionService.Companion.SECRET_KEY
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.ext.Provider
+
+
+@Component
+@Provider
+class AuthenticationFilter : ContainerRequestFilter {
+    companion object {
+        private const val LOGIN_PATH = "jwt/users/login"
+        const val COOKIE_NAME = "ron_cookie_name"  // Replace with actual cookie name
+    }
+
+    override fun filter(requestContext: ContainerRequestContext) {
+        if (requestContext.uriInfo.path == LOGIN_PATH) return
+
+        // Verify the JWT token from the cookie
+        val cookie = requestContext.cookies[COOKIE_NAME]?.value
+
+        try {
+            val jwtClaims = verify(cookie) ?: throw Exception("Invalid token")
+        } catch (e: Exception) {
+            // Catch any exceptions (including null or verification failure) and abort with UNAUTHORIZED status
+            requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).entity("Invalid or expired token").build())
+        }
```

Comment:
i would prefer that all handling of the cookie be done in the verify method. That way this method is clean and doesn't need to know the details of the verification process.

Reply:
Refactored authentication filter to move cookie handling into verify method for cleaner separation of concerns.